src/ucp/core/ucp_worker.c

```diff
@@ -1312,7 +1312,60 @@ static void ucp_worker_init_cpu_atomics(ucp_worker_h worker)
     }
 }
 
-static void ucp_worker_init_device_atomics(ucp_worker_h worker)
+static int ucp_resources_is_device_shared(const ucp_tl_resource_desc_t *rsc1,
+                                          const ucp_tl_resource_desc_t *rsc2)
+{
+    return (rsc1->md_index == rsc2->md_index) &&
+           !strncmp(rsc1->tl_rsc.dev_name, rsc2->tl_rsc.dev_name,
+                    UCT_DEVICE_NAME_MAX);
+}
+
+static ucs_status_t
+ucp_worker_create_loopback_amo_ep(ucp_worker_h worker,
```

Comment:
the name of this function implies that it will return a new ep (either in return or in an arg)